src/harvesters/SaddlePoolManager.sol

```diff
@@ -0,0 +1,57 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.15;
+
+import "BoringSolidity/interfaces/IERC20.sol";
+
+import "../external/saddle/ISwap.sol";
+
+contract SaddlePoolManager {
+    function addLiquidity(
+        address swap,
+        uint256 amountIn,
+        uint256 minToMint
+    ) public returns (uint256 LPTokens) {
+        ISwap hopSwap = ISwap(swap);
+        IERC20 token0 = IERC20(hopSwap.getToken(0));
+        uint256 tokensIn = amountIn;
+
+        if (amountIn == 0) {
+            tokensIn = token0.balanceOf(address(this)) - 1;
+        }
+
+        token0.approve(swap, tokensIn);
+
+        uint256[] memory amounts = new uint256[](2);
+        amounts[0] = tokensIn;
+        amounts[1] = 0;
+
+        LPTokens = hopSwap.addLiquidity(amounts, minToMint, block.timestamp);
+    }
+
+    function swapAndAddLiquidity(
```

Comment:
most comment above apply to this function as well.

You could even call it, to reduce the code